File: fixtures/projects/e2e-python/src/payments/stripe.py
import hashlib

from src.payments.processor import PaymentProcessor, PaymentReceipt, PaymentRequest


class StripeProcessor(PaymentProcessor):
    provider = "stripe"

    def __init__(self, api_key: str) -> None:
        self.api_key = api_key

    def _build_reference(self, request: PaymentRequest) -> str:
        seed = f"{request.reference}:{request.amount}:{self.api_key[:4]}"
        return hashlib.md5(seed.encode("utf-8")).hexdigest()

    def process_payment(self, request: PaymentRequest) -> PaymentReceipt:
        reference = self._build_reference(request)
        approved = request.amount > 0 and request.currency in {"USD", "EUR"}
        return PaymentReceipt(
            provider=self.provider, reference=reference, approved=approved
        )


def build_default_processor() -> StripeProcessor:
    return StripeProcessor(api_key="sk_test_fixture")

File: fixtures/projects/e2e-app/src/payments/stripe.ts
import { PaymentProcessor, type PaymentReceipt, type PaymentRequest } from "./processor";

export class StripeProcessor extends PaymentProcessor {
	readonly provider = "stripe";

	async processPayment(request: PaymentRequest): Promise<PaymentReceipt> {
		const amount = this.normalizeAmount(request.amount);
		const approved = request.currency === "USD" || request.currency === "EUR";
		return {
			id: `stripe-${request.reference}`,
			provider: this.provider,
			approved,
			amount,
			currency: request.currency,
		};
	}

	async refundPayment(receipt: PaymentReceipt): Promise<boolean> {
		return receipt.provider === this.provider && receipt.approved;
	}
}